auth/util/src/test/java/org/wildfly/security/auth/util/MappedRegexRealmMapperTest.java

```diff
@@ -60,4 +62,26 @@ public void testReturnSpecificRealmMapping() {
         Principal principal = new NamePrincipal("A-9999");
         Assert.assertEquals(realmName, mappedRegexRealmMapper.getRealmMapping(principal, null));
     }
+
+    @Test
+    public void testReturnSpecificRealmMappingForDelegate() {
+        String delegateName = "delegate_realm";
+        Pattern pattern = Pattern.compile("([A-Z]-\\d{4})");
+        Map<String, String> realmNameMap = Collections.singletonMap("A-9999", delegateName);
+        MappedRegexRealmMapper mappedRegexRealmMapper = new MappedRegexRealmMapper(pattern, RealmMapper.single("A-9999"), realmNameMap);
+        Principal principal = new NamePrincipal("A-1");
+        Assert.assertEquals(delegateName, mappedRegexRealmMapper.getRealmMapping(principal, null));
+    }
+
+    @Test
+    public void testReturnCorrectRealmMappingWithPrincipalAndDelegateMatchingPattern() {
+        String realmName = "realm";
+        Pattern pattern = Pattern.compile("([A-Z]-\\d{4})");
+        Map<String, String> realmNameMap = new HashMap<>();
+        realmNameMap.put("A-9999", realmName);
+        realmNameMap.put("A-1111", "delegateRealm");
+        MappedRegexRealmMapper mappedRegexRealmMapper = new MappedRegexRealmMapper(pattern, RealmMapper.single("A-1111"), realmNameMap);
+        Principal principal = new NamePrincipal("A-9999");
```

Comment:
@Skyllarr I think my intention was to write a test to determine whether it returns `realmName` or `"delegateRealm"` when both match. However, now I am not sure whether that was the correct implementation of my idea. If I update it as you mentioned, then both tests would test the same thing, which we probably do not want.  If this test does not make sense, maybe it would be better to remove it and keep only the first one. What do you think?

Reply:
@xjusko I see, sorry I misunderstood your intention. I think this test is good then. I am approving this PR. Thank you!